Content.Server/Store/Components/StoreComponent.cs

```diff
@@ -0,0 +1,92 @@
+using Content.Shared.FixedPoint;
+using Content.Shared.Store;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype.Set;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype.Dictionary;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+using Robust.Shared.Audio;
+
+namespace Content.Server.Store.Components;
+
+/// <summary>
+/// This component manages a store which players can use to purchase different listings
+/// through the ui. The currency, listings, and categories are defined in yaml.
+/// </summary>
+[RegisterComponent]
+public sealed class StoreComponent : Component
+{
+    /// <summary>
+    /// The default preset for the store. Is overriden by default values specified on the component.
+    /// </summary>
+    [DataField("preset", customTypeSerializer: typeof(PrototypeIdSerializer<StorePresetPrototype>))]
+    public string? Preset;
+
+    /// <summary>
+    /// All the listing categories that are available on this store.
+    /// The available listings are partially based on the categories.
+    /// </summary>
+    [DataField("categories", customTypeSerializer: typeof(PrototypeIdHashSetSerializer<StoreCategoryPrototype>))]
+    public HashSet<string> Categories = new();
+
+    /// <summary>
+    /// The total amount of currency that can be used in the store.
+    /// The string represents the ID of te currency prototype, where the
+    /// float is that amount.
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite), DataField("balance", customTypeSerializer: typeof(PrototypeIdDictionarySerializer<FixedPoint2, CurrencyPrototype>))]
+    public Dictionary<string, FixedPoint2> Balance = new();
+
+    /// <summary>
+    /// The list of currencies that can be inserted into this store.
+    /// </summary>
+    [ViewVariables(VVAccess.ReadOnly), DataField("currencyWhitelist", customTypeSerializer: typeof(PrototypeIdHashSetSerializer<CurrencyPrototype>))]
+    public HashSet<string> CurrencyWhitelist = new();
+
+    /// <summary>
+    /// Whether or not this store can be activated by clicking on it (like an uplink)
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite), DataField("activateInHand")]
+    public bool ActivateInHand = true;
```

Comment:
instead of having this why not just specify `ActivatableUI` in YAML?

Reply:
This needs to be here because the stores have to use their unique logic for getting all of the listings and such when opened. I'm not sure if that can be done through the generic ActivatableUI component. This is also how the original uplinks did it as well.

Reply:
wait i'm actually dumb lemme just implement this